src/csharp/Microsoft.Spark/Sql/Streaming/DataStreamReader.cs

```diff
@@ -30,6 +31,24 @@ public DataStreamReader Format(string source)
             return this;
         }
 
+        /// <summary>
+        /// Specifies the schema by using <see cref="StructType"/>.
+        /// </summary>
+        /// <remarks>
+        /// Some data sources (e.g. JSON) can infer the input schema automatically
+        /// from data. By specifying the schema here, the underlying data source can
+        /// skip the schema inference step, and thus speed up data loading.
+        /// </remarks>
+        /// <param name="schema">The input schema</param>
+        /// <returns>This DataStreamReader object</returns>
+        public DataStreamReader Schema(StructType schema)
+        {
+            var jvmSchema = (JvmObjectReference)_jvmObject.Jvm.CallStaticJavaMethod(
```

Comment:
How about we create:
```C#
internal static JvmObjectReference FromJson(IJvmBridge jvm, string json)
```
inside `DataType.cs`?

Can you also search for `org.apache.spark.sql.types.DataType`? I see two other places where `fromJson` is called.

Reply:
In `org.apache.spark.sql.types.DataType`, `fromJson` is called [here](https://github.com/apache/spark/blob/fe1f456b200075bef476b5d8eca2be505b486033/sql/catalyst/src/main/scala/org/apache/spark/sql/types/DataType.scala#L129). And there is another `fromJson` is called from `org.apache.spark.sql.types.Metadata` [here](https://github.com/apache/spark/blob/86b25c43506a0a858601bd1082e9f96bb0415eb8/sql/catalyst/src/main/scala/org/apache/spark/sql/types/Metadata.scala#L129). I think the rest `fromJson` methods are not related.

Reply:
oh I meant from our C# repo.

Reply:
Check `UserDefinedFunction.cs` and `UDFRegistration.cs`

Reply:
oh I see. Yes, one is from [here](https://github.com/dotnet/spark/blob/815c2a9a3e4dbbf0004279d07672710948e76d32/src/csharp/Microsoft.Spark/Sql/Expressions/UserDefinedFunction.cs#L37) in `UserDefinedFunction.cs` and the other is from [here](https://github.com/dotnet/spark/blob/815c2a9a3e4dbbf0004279d07672710948e76d32/src/csharp/Microsoft.Spark/Sql/UDFRegistration.cs#L270) in `UDFRegistration.cs`.

Reply:
Added `FromJson` inside `DataType.cs`. Thanks.

Reply:
those two files should use `DataType.FromJson`.

Reply:
Updated.